src/main/java/seedu/badmaths/NotesFileParser.java

```diff
@@ -0,0 +1,116 @@
+package seedu.badmaths;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.Set;
+import java.util.Scanner;
+import java.util.TimerTask;
+import java.util.Timer;
+
+public class NotesFileParser {
+    public static ArrayList<Note> loadFile(String path) {
+        ArrayList<Note> notes = new ArrayList<>();
+        try {
+            Scanner scanner = new Scanner(new File(path));
+            String noteScanner;
+            //text format: Priority | Status | Count | Note
+            while (scanner.hasNextLine()) {
+                noteScanner = scanner.nextLine();
+                String[] noteInfo = noteScanner.split("\\t");
+
+                if (noteInfo.length != 4) {
+                    throw new InvalidFormatException();
+                }
+
+                //priority
+                String priorityStr = noteInfo[0];
```

Comment:
Consider using constants such as `noteInfo[PriorityIndex]` where `private static final int PRIORITY_INDEX = 0; Also, try to refactor the code.